src/ir/term/text/mod.rs

```diff
@@ -440,11 +451,9 @@ impl<'src> IrInterp<'src> {
                             3,
                             "A set_default_modulus should have 2 arguments: modulus and term"
                         );
-                        let m = Arc::new(self.int(&tts[1]));
-                        self.moduli.push(m);
-                        let t = self.term(&tts[2]);
-                        self.moduli.pop().unwrap();
-                        t
+                        let m = self.int(&tts[1]);
+                        self.default_modulus = Some(m);
+                        self.term(&tts[2])
```

Comment:
I'm sorry Riad, I gave you bad information before. This was **not** a bug. The intended behavior is for `set_default_modulus` to be a *term* form that sets the default modulus within the scope of *that term*. So the pop is intentional.

I've pushed a commit that adds a note to the module documentation, reverts the change, and tests the right behavior.

On the other hand, if you think the new behavior is preferable, I'm happy to discuss.

Also, I like the Arc cache :)

Reply:
Edit: While I thought I replied before saying this was a bug, apparently I didn't actually do that? Anyways, I'll just see myself out on my two left feet...

Reply:
Cool, no worries. I'll see what's up with the failures.